src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -485,14 +487,32 @@ public static string GetMSBuildFilePath(PackageSpec project, RestoreRequest requ
                 buildPropsGroup.Position = 2;
 
                 buildPropsGroup.Items.AddRange(sortedPackages.SelectMany(pkg =>
-                    pkg.Key.Build.WithExtension(PropsExtension)
-                        .Where(e => pkg.Value.Exists())
-                        .Select(e => pkg.Value.GetAbsolutePath(e)))
-                        .Select(path => GetPathWithMacros(path, repositoryRoot))
-                        .Select(GenerateImport));
+                        pkg.Key.Build.WithExtension(PropsExtension)
+                            .Where(e => pkg.Value.Exists())
+                            .Select(e => pkg.Value.GetAbsolutePath(e)))
+                    .Select(path => GetPathWithMacros(path, repositoryRoot))
+                    .Select(GenerateImport));
 
                 props.AddRange(GenerateGroupsWithConditions(buildPropsGroup, isMultiTargeting, frameworkConditions));
 
+                // Create an empty PropertyGroup for package properties
+                var packagePathsPropertyGroup = MSBuildRestoreItemGroup.Create("PropertyGroup", Enumerable.Empty<XElement>(), 1000, isMultiTargeting ? frameworkConditions : Enumerable.Empty<string>());
```

Comment:
Anything preventing you from using the same pattern as the build props code?

Reply:
This one is a little different in that its a property group with conditions vs an import group with no conditions containing imports with conditions.  But if you feel strongly I can try to follow a pattern

Reply:
Not sure if I understand correctly but we already add PropertyGroup with condition in nuget.g.props file. wouldn't this be similar to that? We can have a dict of packages path and add them in one shot.